### ISSUE TITLE
number of trials is less than number of possible combinations of the hyper parameters

### ISSUE THREAD

**Issue by shomedas** [keras-team/keras-tuner#417]
I am trying to tune a cnn the hyperparameters of which are listed below 

Conv2d layer(number of filters = 32 or 64 or 96, kernel size = (3*3) or (5*5) or (7*7) )

learning rate : 1e-3 or 1e-5

The total number of combination of hyper parameters are thus 3*3*2=18. When I run the following code i get the result from only 14 trials. I have set the number of max trials to 18.

**My question iis why the number of trials(14) is not equal to the number of combination of the hyper parameters(18)** 

My model is as below

`def build_model(hp):  # random search passes this hyperparameter() object 
    model = keras.models.Sequential()
    
    model.add(Conv2D(hp.Int('input_units',
                                min_value=32,
                                max_value=96,
                                step=32), 
                                hp.Int('kernel_size', 3, 7, 2), 
                                input_shape=x_train.shape[1:]))
    
    model.add(Activation('relu'))
    model.add(MaxPooling2D(pool_size=(2, 2)))

    model.add(Flatten()) 
    model.add(Dense(10))
    model.add(Activation("softmax"))

    model.compile(optimizer=keras.optimizers.Adam(
            hp.Choice('learning_rate',
                      values=[1e-3, 1e-5])),
                  loss="sparse_categorical_crossentropy",
                  metrics=["accuracy"])
    
    return model

tuner = RandomSearch(
    build_model,
    objective='val_accuracy',
    max_trials=18,  # how many model variations to test?
    executions_per_trial=1,  
    directory=LOG_DIR)

tuner.search(x=x_train,
             y=y_train,
             verbose=2, 
             epochs=1,
             batch_size=64,
             #callbacks=[tensorboard],  
             validation_data=(x_test, y_test))
`

The output of keras tuner with 14 entries is as below:

`60000/60000 - 5s - loss: 9.1068 - accuracy: 0.5304 - val_loss: 4.2321 - val_accuracy: 0.6623
[Trial complete]
[Trial summary]
 |-Trial ID: 1d9459631afa1745cceb48dbcfded00c
 |-Score: 0.6622999906539917
 |-Best step: 0
 > Hyperparameters:
 |-input_units: 32
 |-kernel_size: 5
 |-learning_rate: 1e-05
Train on 60000 samples, validate on 10000 samples
60000/60000 - 5s - loss: 1.4349 - accuracy: 0.8351 - val_loss: 0.4025 - val_accuracy: 0.8570
[Trial complete]
[Trial summary]
 |-Trial ID: 4a9595e50117993b19993554ddac6fd7
 |-Score: 0.8569999933242798
 |-Best step: 0
 > Hyperparameters:
 |-input_units: 64
 |-kernel_size: 3
 |-learning_rate: 0.001
Train on 60000 samples, validate on 10000 samples
60000/60000 - 4s - loss: 3.9097 - accuracy: 0.5884 - val_loss: 1.8292 - val_accuracy: 0.7194
[Trial complete]
[Trial summary]
 |-Trial ID: 83791adc900badfdec3c958cd78e0110
 |-Score: 0.7193999886512756
 |-Best step: 0
 > Hyperparameters:
 |-input_units: 64
 |-kernel_size: 7
 |-learning_rate: 1e-05
Train on 60000 samples, validate on 10000 samples
60000/60000 - 5s - loss: 1.0306 - accuracy: 0.8401 - val_loss: 0.4129 - val_accuracy: 0.8612
[Trial complete]
[Trial summary]
 |-Trial ID: 5708ce755a9ed997472588c1875ec6f2
 |-Score: 0.8611999750137329
 |-Best step: 0
 > Hyperparameters:
 |-input_units: 96
 |-kernel_size: 5
 |-learning_rate: 0.001
Train on 60000 samples, validate on 10000 samples
60000/60000 - 4s - loss: 7.5040 - accuracy: 0.5091 - val_loss: 3.1846 - val_accuracy: 0.6746
[Trial complete]
[Trial summary]
 |-Trial ID: f0a7301108b4cceef37318b3f20d1fc8
 |-Score: 0.6746000051498413
 |-Best step: 0
 > Hyperparameters:
 |-input_units: 32
 |-kernel_size: 7
 |-learning_rate: 1e-05
Train on 60000 samples, validate on 10000 samples
60000/60000 - 4s - loss: 3.8665 - accuracy: 0.6088 - val_loss: 1.7027 - val_accuracy: 0.7334
[Trial complete]
[Trial summary]
 |-Trial ID: 4086147064f8b1f3fd574b08b7d83cae
 |-Score: 0.7333999872207642
 |-Best step: 0
 > Hyperparameters:
 |-input_units: 64
 |-kernel_size: 5
 |-learning_rate: 1e-05
Train on 60000 samples, validate on 10000 samples
60000/60000 - 4s - loss: 1.1698 - accuracy: 0.8297 - val_loss: 0.4079 - val_accuracy: 0.8602
[Trial complete]
[Trial summary]
 |-Trial ID: 8ba583da7b9f54c77ceb07c48b65d9e5
 |-Score: 0.8601999878883362
 |-Best step: 0
 > Hyperparameters:
 |-input_units: 64
 |-kernel_size: 5
 |-learning_rate: 0.001
Train on 60000 samples, validate on 10000 samples
60000/60000 - 4s - loss: 1.1080 - accuracy: 0.8317 - val_loss: 0.4387 - val_accuracy: 0.8523
[Trial complete]
[Trial summary]
 |-Trial ID: 7bffe9faf74740fbd3060b2814f3ab1f
 |-Score: 0.8522999882698059
 |-Best step: 0
 > Hyperparameters:
 |-input_units: 32
 |-kernel_size: 3
 |-learning_rate: 0.001
Train on 60000 samples, validate on 10000 samples
60000/60000 - 4s - loss: 8.3851 - accuracy: 0.4636 - val_loss: 3.2361 - val_accuracy: 0.6419
[Trial complete]
[Trial summary]
 |-Trial ID: 72b65b1f2e6f28dc4ddfe7f237f759c1
 |-Score: 0.6419000029563904
 |-Best step: 0
 > Hyperparameters:
 |-input_units: 32
 |-kernel_size: 3
 |-learning_rate: 1e-05
Train on 60000 samples, validate on 10000 samples
60000/60000 - 5s - loss: 1.3747 - accuracy: 0.8451 - val_loss: 0.3648 - val_accuracy: 0.8750
[Trial complete]
[Trial summary]
 |-Trial ID: 3606780412dd910f3caba24e8471a14f
 |-Score: 0.875
 |-Best step: 0
 > Hyperparameters:
 |-input_units: 96
 |-kernel_size: 3
 |-learning_rate: 0.001
Train on 60000 samples, validate on 10000 samples
60000/60000 - 4s - loss: 2.8324 - accuracy: 0.6115 - val_loss: 1.3825 - val_accuracy: 0.7446
[Trial complete]
[Trial summary]
 |-Trial ID: 618d2b1808da658b814effd7ab7cb685
 |-Score: 0.7445999979972839
 |-Best step: 0
 > Hyperparameters:
 |-input_units: 64
 |-kernel_size: 3
 |-learning_rate: 1e-05
Train on 60000 samples, validate on 10000 samples
60000/60000 - 4s - loss: 1.2664 - accuracy: 0.8141 - val_loss: 0.4565 - val_accuracy: 0.8448
[Trial complete]
[Trial summary]
 |-Trial ID: 0ad7bfad6b555aa9c974ed6c4b340527
 |-Score: 0.8447999954223633
 |-Best step: 0
 > Hyperparameters:
 |-input_units: 32
 |-kernel_size: 5
 |-learning_rate: 0.001
Train on 60000 samples, validate on 10000 samples
60000/60000 - 4s - loss: 2.7693 - accuracy: 0.6558 - val_loss: 1.3213 - val_accuracy: 0.7592
[Trial complete]
[Trial summary]
 |-Trial ID: c00c20350f4d2603a3c95c218ddf403d
 |-Score: 0.7591999769210815
 |-Best step: 0
 > Hyperparameters:
 |-input_units: 96
 |-kernel_size: 5
 |-learning_rate: 1e-05
Train on 60000 samples, validate on 10000 samples
60000/60000 - 4s - loss: 1.0968 - accuracy: 0.8244 - val_loss: 0.4446 - val_accuracy: 0.8486
[Trial complete]
[Trial summary]
 |-Trial ID: ef86c3c83841fd2618049efa310dcd20
 |-Score: 0.8485999703407288
 |-Best step: 0
 > Hyperparameters:
 |-input_units: 64
 |-kernel_size: 7
 |-learning_rate: 0.001
{'input_units': 96, 'kernel_size': 3, 'learning_rate': 0.001}
`
